src/plugins/interactive_setup/server/kibana_config_writer.ts

```diff
@@ -54,31 +56,39 @@ export class KibanaConfigWriter {
   public async writeConfig(params: WriteConfigParameters) {
     const caPath = path.join(path.dirname(this.configPath), `ca_${Date.now()}.crt`);
 
-    this.logger.debug(`Writing CA certificate to ${caPath}.`);
-    try {
-      await fs.writeFile(caPath, params.ca);
-      this.logger.debug(`Successfully wrote CA certificate to ${caPath}.`);
-    } catch (err) {
-      this.logger.error(
-        `Failed to write CA certificate to ${caPath}: ${getDetailedErrorMessage(err)}.`
-      );
-      throw err;
+    if (params.caCert) {
+      this.logger.debug(`Writing CA certificate to ${caPath}.`);
+      try {
+        await fs.writeFile(caPath, params.caCert);
+        this.logger.debug(`Successfully wrote CA certificate to ${caPath}.`);
+      } catch (err) {
+        this.logger.error(
+          `Failed to write CA certificate to ${caPath}: ${getDetailedErrorMessage(err)}.`
+        );
+        throw err;
+      }
     }
 
     this.logger.debug(`Writing Elasticsearch configuration to ${this.configPath}.`);
     try {
+      const config: Record<string, any> = { 'elasticsearch.hosts': [params.host] };
+      if (params.username) {
+        config['elasticsearch.username'] = params.username;
+      }
+      if (params.password) {
+        config['elasticsearch.password'] = params.password;
+      }
+      if (params.serviceAccountToken) {
+        config['elasticsearch.serviceAccountToken'] = params.serviceAccountToken.value;
+      }
+      if (params.caCert) {
+        config['elasticsearch.ssl.certificateAuthorities'] = [caPath];
+      }
       await fs.appendFile(
         this.configPath,
-        `\n\n# This section was automatically generated during setup (service account token name is "${
-          params.serviceAccountToken.name
```

Comment:
I removed it for simplicity since service token isn't always going to be present and I didn't see enough value that justified adding a conditionally rendered comment and unit tests for this. Do you think it's critical? Users should be able to delete service tokens using ES APIs or eventually Kibana UI.

Reply:
> Do you think it's critical?

No, I don't think it's critical, we can always change that if it becomes a problem, feel free to keep it as is in your PR.

> Users should be able to view and delete service tokens using ES APIs or eventually Kibana UI.

Correct, that's why I wanted them to know the name of the token, otherwise they won't be able to figure out what token is what (you need to know the name to delete the correct one). But, as I as said above, it's not a big deal.

Reply:
Cool, let's leave it as is for now